Fails to build on s390x with `error[E0308]: mismatched types`

@Joining7943 seems that it is caused by c94a039358?

```

   Compiling uucore v0.0.17 (/<<PKGBUILDDIR>>/src/uucore)
error[E0308]: mismatched types
   --> src/uucore/src/lib/features/fsext.rs:624:16
    |
616 |     fn block_size(&self) -> i64 {
    |                             --- expected `i64` because of return type
...
624 |         return self.f_bsize;
    |                ^^^^^^^^^^^^ expected `i64`, found `u32`
    |
help: you can convert a `u32` to an `i64`
    |
624 |         return self.f_bsize.into();
    |                            +++++++

error[E0308]: mismatched types
   --> src/uucore/src/lib/features/fsext.rs:686:16
    |
678 |     fn fs_type(&self) -> i64 {
    |                          --- expected `i64` because of return type
...
686 |         return self.f_type;
    |                ^^^^^^^^^^^ expected `i64`, found `u32`
    |
help: you can convert a `u32` to an `i64`
    |
686 |         return self.f_type.into();
    |                           +++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `uucore` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
```

Full log:
https://buildd.debian.org/status/fetch.php?pkg=rust-coreutils&arch=s390x&ver=0.0.17-1&stamp=1674426143&raw=0

--- a/src/uucore/src/lib/features/fs.rs
+++ b/src/uucore/src/lib/features/fs.rs
@@ -115,6 +115,7 @@ impl FileInformation {
             not(target_os = "android"),
             not(target_os = "freebsd"),
             not(target_arch = "aarch64"),
+            not(target_arch = "riscv64"),
             target_pointer_width = "64"
         ))]
         return self.0.st_nlink;
@@ -125,6 +126,7 @@ impl FileInformation {
                 target_os = "android",
                 target_os = "freebsd",
                 target_arch = "aarch64",
+                target_arch = "riscv64",
                 not(target_pointer_width = "64")
             )
         ))]

--- a/src/uucore/src/lib/features/fsext.rs
+++ b/src/uucore/src/lib/features/fsext.rs
@@ -619,6 +619,7 @@ impl FsMeta for StatFs {
             not(target_vendor = "apple"),
             not(target_os = "android"),
             not(target_os = "freebsd"),
+            not(target_arch = "s390x"),
             target_pointer_width = "64"
         ))]
         return self.f_bsize;
@@ -626,6 +627,7 @@ impl FsMeta for StatFs {
             not(target_env = "musl"),
             not(target_os = "freebsd"),
             any(
+                target_arch = "s390x",
                 target_vendor = "apple",
                 target_os = "android",
                 not(target_pointer_width = "64")
@@ -681,6 +683,7 @@ impl FsMeta for StatFs {
             not(target_vendor = "apple"),
             not(target_os = "android"),
             not(target_os = "freebsd"),
+            not(target_arch = "s390x"),
             target_pointer_width = "64"
         ))]
         return self.f_type;
@@ -690,6 +693,7 @@ impl FsMeta for StatFs {
                 target_vendor = "apple",
                 target_os = "android",
                 target_os = "freebsd",
+                target_arch = "s390x",
                 not(target_pointer_width = "64")
             )
         ))]

